### ISSUE TITLE
Encoding issue breaks admin sorting

### ISSUE THREAD

**Issue by Swahjak** [Smile-SA/elasticsuite#1703]
### Preconditions

Magento Version : 2.3.3 / 2.3.4

ElasticSuite Version : 2.8.3 / 2.8.4

Environment : Both

Third party modules :

### Steps to reproduce

1. Create a dropdown attribute
2. Set sorting to `_manual`
3. Add attribute option labels with special characters (example: "> 1")

### Expected result

1. Attributes are sorted the same way they are in the admin 'grid'

### Actual result

1. Attributes with special characters are not sorted correctly

This is related to the attribute option label being encoded at https://github.com/Smile-SA/elasticsuite/blob/2.8.3/src/module-elasticsuite-catalog/Model/Layer/Filter/Attribute.php#L217 after that the `isset($items[$optionLabel])` won't pass because the encoded value is `&gt; 1` while the $items value will be `> 1`

**Comment by Swahjak** [comment]
On a side note https://github.com/Smile-SA/elasticsuite/blob/2.8.x/src/module-elasticsuite-catalog/Model/Layer/Filter/Attribute.php#L220 `$optionLabel` will never be null since it's cast to string on line 217

**Comment by Swahjak** [comment]
Dug a little deeper. Issue was introduced with #1039 to fix issue #1037, which it did, but introduced this issue. Would a simple `str_replace(["'", "\""], ["&#039;", "&quot;"])` be enough to fix this issue, or am I missing something?

**Comment by Swahjak** [comment]
@romainruaud (as the original author of #1039). I'm unable to see what the fix in #1039 actually does. I've been testing with some special characters and it doesn't seem to work the way you might expect.

Given a value of `2"` (double quote) it won't pass `isset($items[$optionLabel])`
Given a value of `2''` (2 x single quote) it won't pass `isset($items[$optionLabel])`
Given a value of `> 2` it won't pass `isset($items[$optionLabel])`

So is there any function to this at all? Of has this been 'reintroduced' due to some other changes? Reverting to something like:

```php
            if (!empty($options)) {
                foreach ($options as $option) {
                    if (isset($option['label']) && !empty($option['label'])) {
                        $optionLabel = trim((string) $option['label']);
                        $optionPosition++;

                        if (isset($items[$optionLabel])) {
                            $items[$optionLabel]['adminSortIndex'] = $optionPosition;
                            $items[$optionLabel]['value']          = $optionLabel;
                        }
                    }
                }
            }
```
seems to fix all issues for us.

**Comment by romainruaud** [comment]
I'd say that the proper scenario for testing this is the issue that was occuring in #1037 : 

Creating several option values with quotes and doubles quotes in it, set the attribute to "Manual Sort" (very important, in case of manual sort, we retrieve options from the DB instead of fully trusting the Elasticsearch response), and check if the sort defined in admin is still applied.

Regards

**Comment by Swahjak** [comment]
That is what I've done and so far I don't see any issues (with the suggested code).

The `$items` array does not seem to have any values that are encoded. Maybe I'm overlooking something. Are the database values expected to be encoded in any way? Or is it the ES values that are expected to be encoded? In our case it's neither, but then again I'm going to assume that #1037 was there for a reason.

**Comment by romainruaud** [comment]
`$options` here are coming from `$options = $this->getAttributeModel()->getFrontend()->getSelectOptions();` which is the legacy way to retrieve them.

`$items` is built according the Elasticsearch response directly.

As far as I remember, I had to do the trim/htmlentities thing to be able to have this working : `isset($items[$optionLabel])`

But maybe it's not necessary anymore, it might be due to the way Elasticsearch is encoding (or not) the response he gives. According to the PR date (2018), things might have changed in ES 6.x.

By the way, which version of Elasticsearch are you using ?

**Comment by Swahjak** [comment]
```json
{
  "name" : "DyrUsQ9",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "m3cEaUgSSA2gFu4D25Qvnw",
  "version" : {
    "number" : "6.4.2",
    "build_flavor" : "default",
    "build_type" : "tar",
    "build_hash" : "04711c2",
    "build_date" : "2018-09-26T13:34:09.098244Z",
    "build_snapshot" : false,
    "lucene_version" : "7.4.0",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by Swahjak** [comment]
Created a pull which we will use to patch our issue for now. Please let me know if there are any issues / if I've overlooked something.

**Comment by rbayet** [comment]
Fixed by proposed #1705
